community/modules/compute/gke-node-pool/main.tf

```diff
@@ -158,6 +169,12 @@ resource "google_container_node_pool" "node_pool" {
       condition     = !(var.local_ssd_count_ephemeral_storage > 0 && var.local_ssd_count_nvme_block > 0)
       error_message = "Only one of local_ssd_count_ephemeral_storage or local_ssd_count_nvme_block can be set to a non-zero value."
     }
+
+    # non preattached gpu machine type should always have guest_accelerator config defined
+    precondition {
+      condition     = (local.preattached_gpu_machine_family && length(var.guest_accelerator != null ? var.guest_accelerator : []) >= 0) || (!local.preattached_gpu_machine_family && length(var.guest_accelerator != null ? var.guest_accelerator : []) > 0)
```

Comment:
`!local.preattached_gpu_machine_family && length(var.guest_accelerator != null ? var.guest_accelerator : []) > 0`

I think this condition is incorrect, how about for `machine_type: c2-standard-60`. I would expect `var.guest_accelerator` to be `null`.

Eventually, the error message below is not true. I don't think we can actually do validation here because regardless of whether or not `preattached_gpu_machine_family`, we may or may not have `guest_accelerator`.

Reply:
Hmm, makes sense for the first validation, I think added that as for non preattached gpu families the second condition was becoming false but I agree it can be `local.preattached_gpu_machine_family || (!local.preattached_gpu_machine_family && length(var.guest_accelerator != null ? var.guest_accelerator : []) > 0)`

Additionally I had an assumption that the user will either use a pre-attached gpu family machine type or add `guest_accelerator` but if that assumption is incorrect, there is no point of adding this validation here and instead focus on making the defaults / optional work better.